packages/forms-web-app/src/controllers/examination/your-interested-party-number.js

```diff
@@ -0,0 +1,72 @@
+const {
+	routesConfig: {
+		examination: {
+			pages: {
+				yourInterestedPartyNumber: {
+					view: yourInterestedPartyNumberView,
+					route: yourInterestedPartyNumberRoute,
+					id: yourInterestedPartyNumberId
+				},
+				haveAnInterestedPartyNumber: { route: haveAnInterestedPartyNumberRoute },
+				submittingFor: { route: submittingForRoute }
+			}
+		}
+	},
+	routesConfig: { examination }
+} = require('../../routes/config');
+
+const config = require('../../config');
+
+const examinationSession = config.sessionStorage.examination;
+const backLinkUrl = `${examination.directory + haveAnInterestedPartyNumberRoute}`;
+const pageTitle = "What's your interested party number?";
+const title = pageTitle;
+
+const getYourInterestedPartyNumber = (req, res) => {
+	const { session = { examination: { name: null } } } = req;
+	const reqExaminationSession = session[examinationSession?.name] ?? '';
+
+	if (!reqExaminationSession) return res.status(404).render('error/not-found');
+
+	const { interestedPartyNumber = '' } = reqExaminationSession;
+
+	res.render(yourInterestedPartyNumberView, {
+		id: yourInterestedPartyNumberId,
+		backLinkUrl,
+		interestedPartyNumber,
+		pageTitle,
+		title
+	});
+};
+
+const postYourInterestedPartyNumber = (req, res) => {
+	const { body = { yourInterestedPartyNumberId: '' } } = req;
```

Comment:
Could have a try/catch here to handle any errors.

Reply:
added for both get and post routes